src/main/java/edu/usf/cutr/gtfs_realtime/bullrunner/GtfsRealtimeProviderImpl.java

```diff
@@ -16,688 +16,457 @@
+                vehicleArray = downloadVehiclesRouteC(route_id, external_id);
+            } else {
+                vehicleArray = downloadVehicles(external_id);
+            }
+
+            // Loop through vehicleArray to build vehiclePosition for the given route
+            HashSet addedVehicleIds = new HashSet(); // keep track of vehicles that have been added to the feed
+            for (int k = 0; k < vehicleArray.length(); k++) {
+                JSONObject vehicleObj = vehicleArray.getJSONObject(k);
+                // check if vehicle has been added to feed
+                if (addedVehicleIds.contains(vehicleObj.getString(FIELD_NAME))) {
+                    continue;
+                } else {
+                    addedVehicleIds.add(vehicleObj.getString(FIELD_NAME));
+                }
+                // check if we need to skip setting route_id if the API returns empty response for route C and MSC
+                boolean has_route_id;
+                if ((route_id.equals("C") || route_id.equals("MSC Express")) &&
+                        vehicleObj.getString("route_id").equals("Unknown")){
+                    has_route_id = false;
+                } else {has_route_id = true;}
+
+                // initiate feed
+                TripDescriptor.Builder tripDescriptor = TripDescriptor.newBuilder();
+                Position.Builder position = Position.newBuilder();
+                VehicleDescriptor.Builder VehicleInfo = VehicleDescriptor.newBuilder();
+                FeedEntity.Builder vehiclePositionEntity = FeedEntity.newBuilder();
+
+                // set values for feed
+                if (has_route_id) tripDescriptor.setRouteId(route_id);
+                if (has_route_id) tripDescriptor.setTripId(findTripID(route_id));
+                position.setBearing((float) Math.round(100 * vehicleObj.getDouble(FIELD_HEADING_DEGREES)) / 100);
+                position.setLatitude((float) vehicleObj.getDouble(FIELD_LAT));
+                position.setLongitude((float) vehicleObj.getDouble(FIELD_LON));
+                position.setSpeed((float) Math.round(100 * vehicleObj.getDouble(FIELD_SPEED)) / 100);
+                VehicleInfo.setId(vehicleObj.getString(FIELD_NAME));
+                VehicleInfo.setLabel(vehicleObj.getString(FIELD_NAME));
+                tripDescriptor.setScheduleRelationship(TripDescriptor.ScheduleRelationship.UNSCHEDULED);
+
+                // Build feed
+                VehiclePosition.Builder vpBuilder = VehiclePosition.newBuilder();
+                vpBuilder.setPosition(position);
+                vpBuilder.setTrip(tripDescriptor);
+                vpBuilder.setVehicle(VehicleInfo);
+                vpBuilder.setTimestamp(
+                        Instant.parse(vehicleObj.getString(FIELD_LAST_UPDATED)).getEpochSecond()
+                );
+                vpBuilder.hasVehicle();
+                if (vehicleObj.getDouble(FIELD_PASSENGER_LOAD) <= 0) {

```

Comment:
@minhhpham Could you take a look at the passenger load values?  It looks like we're always reporting `MANY_SEATS_AVAILABLE`.  I think the value returned from the API may have changed from an integer (which the below IF statements are written for) to a fraction - it looks like this in the new API - `"passengerLoad": 0.11`.